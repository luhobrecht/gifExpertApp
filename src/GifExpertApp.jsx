import React, { useState } from 'react'
import { AddCategory, GifGrid } from './components';

export const GifExpertApp = () => {

    const [categories, setCategories] = useState([ 'Friends' ]);
    
    const onAddCategory = (newCategory) => {

        if(categories.includes(newCategory)) return;
  
        setCategories([ newCategory , ...categories ]);
        //o
        //setCategories( cat => [...cat, 'Peaky Blinders'])

    }

  return (
    <>

   
    <h1>Gif Expert App</h1>

 
    <AddCategory 
    onNewCategory ={ event => onAddCategory(event) }
    />

    

    {categories.map((category) => (
            <GifGrid 
            key={ category }
            category= { category }
            />
        )
    )}
   

    </>
  )
}
