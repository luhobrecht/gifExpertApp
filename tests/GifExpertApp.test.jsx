import { render, screen } from "@testing-library/react"
import { GifExpertApp } from "../src/GifExpertApp"


describe('Pruebas en <GifExpertApp />', () => { 
    
    test('should match the snapshot ', () => { 
        

        const title = "Fuck You Season 4 GIF by Friends"
        const url = "https://media2.giphy.com/media/daIJvyatzeg20Iqi12/giphy.gif?cid=97d5ddfc1osd30iu6jr8822y0r3dmpgrx1lqw7gzxgj4u476&rid=giphy.gif&ct=g"

        const {container} = render( <GifExpertApp title= { title } url= { url }/> )

       expect (container).toMatchSnapshot();
     //screen.debug();

     });

     test('should return false if same category is added', () => { 
         

        // const newCategory = 'Friends'

        // const onAddCategory = render(<GifExpertApp { newCategory }/>)
 
        // expect(screen.getByRole('textbox')).toBeFalsy()


      })

    //  


    })