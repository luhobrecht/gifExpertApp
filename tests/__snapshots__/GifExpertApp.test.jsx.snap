// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> should match the snapshot  1`] = `
<div>
  <h1>
    Gif Expert App
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    Friends
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
